### PR TITLE
14/ClientAI - Network API

### PR DESCRIPTION
**Pull request by LucasTesnier**
Implement the architecture of the network API.

Add login and logout to the network server process.

### CODE REVIEW

**Review by 31Nathan — APPROVED** [review]
LGTM!

File: src/zappy_ai_src/network/entrypoint.c [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+{
+    int port = atoi(av[1]);
+    char *team_name = av[2];
+
+    (void) team_name;
+    (void) ac;
+    signal(SIGINT, sigint_handler);
+    if (!c_interface_try_to_connect_to_server("127.0.0.1", port))
+        return 84;
+    while (1);
```
Comment (Cavonstavant) [inline src/zappy_ai_src/network/entrypoint.c:31]: To be removed for obvious reasons

Comment (LucasTesnier) [reply src/zappy_ai_src/network/entrypoint.c:31]: Read the top of the file :eyes:

File: src/zappy_ai_src/network/net_utils/include/client_utils.h [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    const char *func __attribute__((unused)),
+    const char *msg __attribute__((unused)))
+{
+    char *errmsg = strerror(errno);
+
+    printf("%s:%d:%s():\n\t%s: %s\n", file, line, func, msg, errmsg);
+}
+
+    /// \brief Simple macro used to log a message
+    #define TEAMS_LOG(msg) \
```
Comment (Cavonstavant) [inline src/zappy_ai_src/network/net_utils/include/client_utils.h:78]: o no

File: src/zappy_ai_src/network/net_utils/update_client.c [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+void update_client(client_net_server_t *server)
+{
+    if (check_for_disconnection(server))
+        return;
+    fill_fd_sets(server);
+    if (select(FD_SETSIZE, &server->read_fds,
+        &server->write_fds, NULL, NULL) == -1) {
+        TEAMS_LOG("Internal Server Error: select\n");
```
Comment (Cavonstavant) [inline src/zappy_ai_src/network/net_utils/update_client.c:52]: o no

**Review by Cavonstavant — CHANGES_REQUESTED** [review]

**Pull request merged**